### PR TITLE
Add Rust and Cargo

### PR DESCRIPTION
**Pull request by kcotugno**
Here is a configuration of Rust and Cargo which builds on all four architectures.  

There are two issue currently:
1. `clippy-driver` fails to build on `arm`. This is due to cargo adding both `--target=armv7-linux-android` and `--target=x86_64-linux-android` when building `backtrace-sys`. It is a `C` dependency so these flags are sent to `clang` which cause some bugs in certain header macros. It compiled successfully when manually running the command without the extra flag. I have been unable to figure out both why the same does not happen on `aarch64` and how to remove the extra target flag.
2. `Cargo` requires `USER` environment variable to be set. I took a brief look at the Termux app to see if we can set it when we set up the other environment variable such as the `PATH` to no avail. Maybe we can add `export USER=$(whoami)` to the global profile?

Since these are not breaking issues, we can choose to ship now if we so desire.

I only have an aarch64 phone, so if others can test the other architectures, I would greatly appreciate it.

Here are the links. FYI these were built with an incorrect tmpdir patch.

https://s3-us-west-2.amazonaws.com/axius/rust_1.29.2_aarch64.deb
https://s3-us-west-2.amazonaws.com/axius/rust_1.29.2_arm.deb
https://s3-us-west-2.amazonaws.com/axius/rust_1.29.2_i686.deb
https://s3-us-west-2.amazonaws.com/axius/rust_1.29.2_x86_64.deb

### CODE REVIEW

**Comment by its-pointless** [comment]
If you build clippy-driver for local host it will build for arm.
Its a bug they fixed in nightly.

File: packages/rust/build.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+TERMUX_PKG_VERSION=1.29.2
+TERMUX_PKG_SRCURL=https://static.rust-lang.org/dist/rustc-$TERMUX_PKG_VERSION-src.tar.gz
+TERMUX_PKG_SHA256=5088e796aa2e47478cdf41e7243fc5443fafab0a7c70a11423e57c80c04167c9
+TERMUX_PKG_DEPENDS="clang, openssl"
+
+termux_step_configure () {
+	termux_setup_cmake
+
+	if [ "$TERMUX_ARCH" = "arm" ]; then
+		TRIPLE="armv7-linux-androideabi"
```
Comment (joakim-noah) [inline packages/rust/build.sh]: Current triple is almost same, only missing `v7`, could be fixed [this way instead](https://github.com/termux/termux-packages/blob/master/packages/ldc/build.sh#L40).

File: packages/rust/build.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	else
+		TRIPLE="$TERMUX_HOST_PLATFORM"
+	fi
+
+	sed "s%\\@TERMUX_PREFIX\\@%$TERMUX_PREFIX%g" \
+		$TERMUX_PKG_BUILDER_DIR/config.toml \
+		| sed "s%\\@TERMUX_STANDALONE_TOOLCHAIN\\@%$TERMUX_STANDALONE_TOOLCHAIN%g" \
+		| sed "s%\\@TERMUX_STANDALONE_TOOLCHAIN\\@%$TERMUX_STANDALONE_TOOLCHAIN%g" \
```
Comment (joakim-noah) [inline packages/rust/build.sh]: Same replacement twice.

**Comment by fornwall** [comment]
Nice! The rust package is now available for installation.

**Comment by kcotugno** [comment]
Thank you @fornwall. I'll be opening a PR soon which fixes the ARM build issues and bumps the version to 1.30.0 which was just released just few days ago.

**Pull request merged**